spec.bs

```diff
@@ -595,11 +653,14 @@ directly; for authentication the extension can only be accessed via
     dictionary AuthenticationExtensionsPaymentInputs {
       boolean isPayment;
 
+Q: When would this be set to False? Is this not true by the virtue of being present?
```

Comment:
Yes, this is just an awkward outcome of WebIDL treating an empty dict as false (afaik). So if during registration you did:

```
const publicKeyCreationOptions = {
  ...
  extensions: {
    payment: {},
  },
};
navigator.credentials.create({publicKey: publicKeyCreationOptions});
```

Then afaik the payment extension is actually *not* set. So you need some way to indicate 'no I do want this'.

As per the note under https://w3c.github.io/secure-payment-confirmation/#sctn-enrollment, if WebAuthn WG allow credential creation in a 3p context, and if we get to a point where the browser doesn't have to cache the credentials at creation time, this extension will no longer need to be registration time too, and this field can be dropped.

(`isPayment` definitely needs  better documentation. It is mentioned here https://w3c.github.io/secure-payment-confirmation/#dom-authenticationextensionspaymentinputs-ispayment but not well)